editor/src/core/shared/memoize.ts

```diff
@@ -23,3 +23,30 @@ export function memoize<T extends Moizeable>(func: T, options?: Partial<Options<
   const memoizeOptions = getMemoizeOptions(options)
   return moize(func, memoizeOptions)
 }
+
+export function valueDependentCache<Value, Input, Result>(
+  fallback: (value: Value, input: Input) => Result,
+  inputToString: (input: Input) => string,
+): (value: Value, input: Input) => Result {
+  let cache: { [key: string]: Result } = {}
```

Comment:
maybe consider a `Map` here with arbitrary keys? (and instead the `inputToString` we can have an equality function)